### PR TITLE
Apply missing lr_mult and wd_mult to the lr and weight_decay of megatron param groups.

### PR DESCRIPTION
**Pull request by jstjohn**
**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by JRD971000** [comment]
@marcromeyn @hemildesai Could you please review this? The pytorch schedulers don't include the weight decay keys from megatron lr scheduler which are passed to the optimizer (e.g. no weight decay cond).

**Comment by jstjohn** [comment]
For some more background, if you look through megatron, the only places that lr_mult or wd_mult are applied are in the custom scheduler:
* https://github.com/search?q=repo%3ANVIDIA%2FMegatron-LM%20wd_mult&type=code
* https://github.com/search?q=repo%3ANVIDIA%2FMegatron-LM+lr_mult&type=code

In NeMo we initialze the optimizer with lr_mult like megatron does (just sets the local variable) which I think propagates into the defaults for the param_groups, but from there it's not used in the core optimizer at all, just by this scheduler to update the LR and wd.

**Comment by jstjohn** [comment]
Note that the strategy in this PR is implicitly is `lr_i = schedule(lr_mult*init_lr, i))` while the one in the megatron scheduler is `lr_i = lr_mult*(schedule(init_lr,i))` (and similar for wd)

**Review by ananthsub — APPROVED** [review]
Thanks for catching this! could you also add a unit test that confirms that `setup_megatron_optimizer` now respects these param groups settings?

**Comment by jstjohn** [comment]
> Thanks for catching this! could you also add a unit test that confirms that `setup_megatron_optimizer` now respects these param groups settings?

@ananthsub I added tests that will only pass if this fix is in place to all of our tested models. This should easily catch a regression.

**Review by hemildesai — APPROVED** [review]

**Review by hemildesai — DISMISSED** [review]

**Review by hemildesai — DISMISSED** [review]
